.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,153 @@
+        with:
+          python-version: ${{ needs.python_version.outputs.version }}
+
+      - uses: webfactory/ssh-agent@v0.5.4
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - name: Install poetry
+        run: pipx install poetry
+
+      # Poetry must explicitly be told to use the expected version https://github.com/actions/setup-python/issues/374
+      - name: Install dependencies
+        run: |
+          poetry env use ${{ needs.python_version.outputs.version }}
+          poetry config http-basic.properly "${{ secrets.PIP_GEMFURY_DEPLOY_TOKEN }}" NOPASS
+          poetry install
+          npm install
+
+      - name: Lint
+        run: poetry run inv lint
+
+      - name: Typecheck
+        if: inputs.run_typecheck 
+        run: poetry run inv typecheck
+
+      - name: Run Tests
+        run: poetry run inv test --junitxml="tests/results.xml"
+
+      - name: Publish test results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          files: "${{ inputs.working_directory }}/tests/*.xml"
+          comment_mode: "always"
+
+      - name: Upload metadata to Datadog
+        if: inputs.run_datadog_ci
+        run: poetry run inv datadog-ci
+        env:
+          DATADOG_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Start deployment (staging)
+        id: deployment_staging
+        if: github.ref == 'refs/heads/main' && inputs.deploy_staging
+        uses: bobheadxi/deployments@v1
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: staging
+
+      - name: Create domain (staging)
+        if: github.ref == 'refs/heads/main' && inputs.deploy_staging
+        run: npx sls create_domain --stage staging
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Deploy (staging)
+        if: github.ref == 'refs/heads/main' && inputs.deploy_staging
+        run: poetry run inv deploy --stage staging
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          DD_API_KEY: ${{ secrets.DD_API_KEY }}
+
+      - name: Update deployment status (staging)
+        if: always() && github.ref == 'refs/heads/main' && inputs.deploy_staging
+        uses: bobheadxi/deployments@v1
+        with:
+          step: finish
+          token: ${{ secrets.GITHUB_TOKEN }}
+          status: ${{ job.status }}
+          env: ${{ steps.deployment_staging.outputs.env }}
+          env_url: https://template-python-serverless-fastapi-staging.api.properly.ca/docs
+          deployment_id: ${{ steps.deployment_staging.outputs.deployment_id }}
+
+      - name: Start deployment (prod)
+        id: deployment_prod
+        if: github.ref == 'refs/heads/master'
```

Comment:
We have some branches named `main` and other ones named `master` - what should this be, and does it need to accommodate both?

Reply:
That's a miss! Thanks for pointing it out. I'll have to pass the default branch name as a parameter.